quinn-proto/src/cid_generator.rs

```diff
@@ -0,0 +1,60 @@
+use rand::RngCore;
+
+use crate::shared::ConnectionId;
+use crate::MAX_CID_SIZE;
+
+/// Generates connection IDs for incoming connections
+pub trait ConnectionIdGenerator: Send {
+    /// Generates a new CID
+    ///
+    /// Connection IDs MUST NOT contain any information that can be used by
+    //    an external observer (that is, one that does not cooperate with the
+    //    issuer) to correlate them with other connection IDs for the same
+    //    connection.
+    fn generate_cid(&mut self) -> ConnectionId;
+    /// Performs any validation if it is needed (e.g. HMAC, etc)
+    ///
+    /// Apply validation check on those CIDs that may still exist in hash table
+    ///   but considered invalid by application-layer logic.
+    /// e.g., We may want to limit the amount of time for which a CID is valid
+    ///   in order to reduce the number of valid IDs that could be accumulated
+    ///   by an attacker.
```

Comment:
Stray whitespace. Unless I'm missing something about markdown?
```suggestion
    /// but considered invalid by application-layer logic.
    /// e.g. we may want to limit the amount of time for which a CID is valid
    /// in order to reduce the number of valid IDs that could be accumulated
    /// by an attacker.
```